src/shout_subgroup/utils.py

```diff
@@ -8,3 +7,14 @@ def usernames_valid(usernames: set[str]) -> bool:
     # TODO: Implement stricter validations later
     return all(isinstance(item, str) for item in usernames)
 
+
+async def is_group_chat(telegram_chat_id: int) -> bool:
```

Comment:
why is this function async?

Reply:
Good question.

I wasn't confident the condition would be checked before the business logic here

```
telegram_chat_id = telegram_chat.id
    if not await is_group_chat(telegram_chat_id):
        msg = f"Can't create subgroup because telegram chat id {telegram_chat_id} is not a group chat."
        logging.info(msg)
        raise NotGroupChatError(msg)

    # If the code reaches to this point the telegram_chat_id has to be a telegram group chat id
    subgroup = await find_subgroup_by_telegram_group_chat_id_and_subgroup_name(db, telegram_chat_id, subgroup_name)
    if subgroup:
        msg = f"Subgroup {subgroup_name} already exists for telegram group chat id {telegram_chat_id}."
        logging.info(msg)
        raise SubGroupExistsError(msg)
```

I erred on the side of caution. If we find out `async` doesn't work this way, let's move it back to sync